### PR TITLE
[3.9][a11y] Add landmark to breadcrumbs

### PR DESCRIPTION
**Pull request by wojsmol**
Pull Request for Issue # .

### Summary of Changes
- added nav tag
- added attribute aria-label and role="navigation" to nav tag
### Testing Instructions
- code inspection
- use screen reader (e.g. NVDA) and see that the breadcrumbs is accessible during navigation as a landmark 

Similar to  PR https://github.com/joomla/joomla-cms/pull/23685 for Joomla 4.
### Expected result
The 'nav' tag defines one of the main landmarks of the page, and the aria-label attribute makes it possible to distinguish this landmark from other navigation points, e.g. general menu.  
Landmarks are supported by assistive technologies. Users of screen readers use them as one of the navigation systems on the page. 
### Actual result
Assistive technology does not recognize the breadcrumb as a landmark.
### Documentation Changes Required
In the Help screens
Include information that the title of the module is used as a label for assistive technologies, so it should be translated (since the title of this module is not displayed, administrators often leave it in English).

### CODE REVIEW

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on 7bf3f2fbb1d79db6c9c2daa6759dfb74e9a8556b<br /><br />In my opinion, the `role="navigation"` attribute is unnecessary. You can leave it. But all popular browsers, including IE11, recognize the `nav` tag. `Nav` tag has `role=navigation` implicite entered. Leaving an attribute, however, will not be an error. In the W3C WAI tutorial, it is still recommended to place a redundant attribute of a role.  
See: https://www.w3.org/WAI/tutorials/page-structure/regions/ :
> Most current web browsers support the above HTML5 elements and convey the information to assistive technology through the accessibility APIs. However, to maximize compatibility with web browsers and assistive technologies that support WAI-ARIA but do not yet support HTML5, it is currently advisable to use both the HTML5 elements and the corresponding WAI-ARIA roles.

Personally, I am against entering a role attribute. But because we did this in other cases (e.g. in the pagination.php file), let it stay.  <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23691">issues.joomla.org/tracker/joomla-cms/23691</a>.</sub>

**Comment by brianteeman** [comment]
Sorry but you don't actually understand what the text you are quoting refers to.

The nav element is not exposed in the accessibility tree with internet explorer. 

What you are seeing is that NVDA (and perhaps other screen readers have written a workaround to this IE bug) See https://github.com/nvaccess/nvda/pull/6044

This only means that an NVDA and IE11 user is helped. It doesn't fix the bug in IE11 which we have no way of knowing if other assistive technology has their own work around.

So the correct conclusion is to have the role so that ALL users of  assistive technology and IE 11 are covered. 

We only support specific browsers. We do not support specific  assistive technology

**Comment by zwiastunsw** [comment]
I am sorry. I don't know why you assume that I don't understand something. Maybe it' s you who don't understand something? 
I wrote what my opinion is. I am free to have my opinion. At the same time I did a successful test. 
Please see the new W3C WAI page. 
https://www.w3.org/WAI/
Please check in the code that the new HTML5 tags (header, nav, main, footer, etc) are used without the role=" attribute. 
Of course, you can say that W3C WAI does wrong. For me this is the correct pattern. 
Using these tags (which have an implicitly entered role) without role attributes is correct (and not, as you say, incorrect). Even if some technologies do not support these tags yet.

**Comment by brianteeman** [comment]
Sorry I can't help it if you don't understand the difference between the html 5 specification and the browsers implementation of it and it's the accessibility html5 mapping specification aka html-aam

**Comment by zwiastunsw** [comment]
@brianteeman believe me, I understand.

**Comment by HLeithner** [comment]
This PR adds a feature and may break existing templates without overrides, so this will likely not go into J3.

There is already a J4 PR for this https://github.com/joomla/joomla-cms/pull/23685 correct.

**Comment by brianteeman** [comment]
@hleithner
For 3.x if it. Is changed to div role=navigation it will not break anything and it will be more accessible than it was before

**Comment by HLeithner** [comment]
This PR adds a NAV-tag around the UL-tag and this can break templates.

**Comment by brianteeman** [comment]
Yes which is why I wrote what I wrote

**Comment by HLeithner** [comment]
Maybe break less but why can't it be added to the UL?

**Comment by brianteeman** [comment]
it can be but it would be invalid html

**Comment by zwiastunsw** [comment]
1. How will adding `nav` tag damage existing templates? The Breadcrumb is stylized with the `breadcrumb` class.  Of course, it is always possible that some inexperienced webmaster will use some special formatting for the 'nav' tag. But we won't be able to do that.
2. Adding an attribute `role="navigation"` to `div` tag  is not an error if you cannot use `nav` tag (for ex. in HTML4 document), but adding to `ul` tag is the error! But there is no justification here that excludes the use of `nav` tag.
3. Let me also remind you:
> 6.1.7 Rendered markup
> For the time being, rendered markup is not subject to our backwards compatibility promise. We will try not to change markup in such a way that a site might render differently, but we can't promise not to break anything at the present time. We will work on defining ways in which we might make a backwards compatibility promise for markup in the future, but we do not currently have a satisfactory consensus on a workable standard.
https://developer.joomla.org/development-strategy.html

File: modules/mod_breadcrumbs/tmpl/default.php [hunk 1]
```diff
@@ -9,65 +9,66 @@
 
 defined('_JEXEC') or die;
 ?>
+<nav aria-label="<?php echo $module->name; ?>" role="navigation">
```
Comment (brianteeman) [inline modules/mod_breadcrumbs/tmpl/default.php]:
```suggestion
<div aria-label="<?php echo $module->name; ?>" role="navigation">
```

File: modules/mod_breadcrumbs/tmpl/default.php [hunk 2]
```diff
@@ -9,65 +9,66 @@
...
-			</li>
-		<?php endif;
-	endforeach; ?>
-</ul>
+					<meta itemprop="position" content="<?php echo $key + 1; ?>">
+				</li>
+			<?php endif;
+		endforeach; ?>
+	</ul>
+</nav>
```
Comment (brianteeman) [inline modules/mod_breadcrumbs/tmpl/default.php]:
```suggestion
</div>
```

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on f57389c4d8de47b8a9a5e977f8c1eec3afdae42f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23691">issues.joomla.org/tracker/joomla-cms/23691</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on f57389c4d8de47b8a9a5e977f8c1eec3afdae42f<br /><br />You cant close a div with a nav :(<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23691">issues.joomla.org/tracker/joomla-cms/23691</a>.</sub>

**Comment by brianteeman** [comment]
@zwiastunsw how did you successfully test - the markup is wrong

**Comment by zwiastunsw** [comment]
Yes, my mistake.

**Comment by wojsmol** [comment]
@brianteeman @zwiastunsw Issue fixed :smile:

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on ede34b477874366a8c1ae6445aa8e934666910a3<br /><br />now is OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23691">issues.joomla.org/tracker/joomla-cms/23691</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on ede34b477874366a8c1ae6445aa8e934666910a3<br /><br />Thats better - thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23691">issues.joomla.org/tracker/joomla-cms/23691</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23691">issues.joomla.org/tracker/joomla-cms/23691</a>.</sub>

**Comment by wojsmol** [comment]
@HLeithner Please merge this PR, `nav` was changed to `div` as requested.

**Comment by HLeithner** [comment]
I requested a different change that seams not to be vaild, thats the reason I didn't merged it.

**Comment by wojsmol** [comment]
@HLeithner This PR now adapt changes suggested ny @brianteeman in https://github.com/joomla/joomla-cms/pull/23691#issuecomment-457949851 see https://github.com/joomla/joomla-cms/pull/23691/files?utf8=%E2%9C%93&diff=unified&w=1

**Comment by HLeithner** [comment]
I merge it anyway, thx

**Pull request merged**